structuredheadings/plugin.js

```diff
@@ -730,31 +729,33 @@
 
           return [CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)];
         },
-
-        exec: function (editor, value) {
-          var selection = editor.getSelection();
-          var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
-                editor,
-                selection
-              );
+        exec: function (editor, value) { // eslint-disable-line max-statements
+          var headingIterator = CKEDITOR.plugins.structuredheadings.getHeadingIteratorForSelection(
```

Comment:
There is a lot of bad naming here.

 Because of how the headingIterator is actually just wrapping the usual iterator with a filter inside a closure, the iterator itself should probably be named something like getNextHeading.  I will rename some things to make it more clear.